internal/controller/api/v1alpha1/agentserviceconfig_types.go

```diff
@@ -24,10 +24,37 @@ import (
 
 // AgentServiceConfigSpec defines the desired state of AgentServiceConfig
 type AgentServiceConfigSpec struct {
+	// FileSystemStorage defines the spec of the PersistentVolumeClaim to be
+	// created for the assisted-service's filesystem (logs, etc).
+	// TODO(djzager) determine method for communicating guidelines
+	// for access modes, resources, and storageClass
 	FileSystemStorage corev1.PersistentVolumeClaimSpec `json:"filesystemStorage"`
-	DatabaseStorage   corev1.PersistentVolumeClaimSpec `json:"databaseStorage"`
+	// DatabaseStorage defines the spec of the PersistentVolumeClaim to be
+	// created for the database's filesystem.
+	// TODO(djzager) determine method for communicating guidelines
+	// for access modes, resources, and storageClass
+	DatabaseStorage corev1.PersistentVolumeClaimSpec `json:"databaseStorage"`
 }
 
+const (
+	// FilesystemStorageCreated reports whether the filesystem PVC has been created.
+	FilesystemStorageCreated conditionsv1.ConditionType = "FilesystemStorageCreated"
+	// DatabaseStorageCreated reports whether the database PVC has been created.
+	DatabaseStorageCreated conditionsv1.ConditionType = "DatabaseStorageCreated"
+	// AgentServiceCreated reports whether the assisted-service service was created.
+	AgentServiceCreated conditionsv1.ConditionType = "AgentServiceCreated"
+	// DatabaseServiceCreated reports whether the database service was created.
+	DatabaseServiceCreated conditionsv1.ConditionType = "DatabaseServiceCreated"
+	// AgentRouteCreated reports whether the assisted-service route was created.
+	AgentRouteCreated conditionsv1.ConditionType = "AgentRouteCreated"
+	// DatabaseSecretCreated reports whether the database secret was created.
+	DatabaseSecretCreated conditionsv1.ConditionType = "DatabaseSecretCreated"
+	// ServiceDeploymentCreated reports whether the assisted-service deployment was created.
+	ServiceDeploymentCreated conditionsv1.ConditionType = "ServiceDeploymentCreated"
+	// DatabaseDeploymentCreated reports whether the database deployment was created.
+	DatabaseDeploymentCreated conditionsv1.ConditionType = "DatabaseDeploymentCreated"
```

Comment:
Many of these seem too granular to me. What would otherwise be implementation details are now elevated to be part of the API. Will we always create a Deployment for the DB? Maybe we'll delegate that to some other operator in the future. Will we always have agents communicate with a Route? etc.

I think it's better to be more qualitative. Is the operand running? Is the operand healthy? That kind of stuff.

Since it's a lot easier to add conditions than take them away, I'd rather start minimal and add later if/when the need arises.

Reply:
I'm not aware of there being any guarantees that conditions and conditionTypes won't change. Could you point me towards something.

In some aspects I believe I see your point (do we actually care about the individual objects and there maybe being too many stand out), but even in the work that I and @rwsu have been doing I've found it pretty helpful for the Conditions to tell me exactly where the problem is. It may be possible to put this all under a `ReconcileCompleted` condition, but before we do that, I'd like to know that there is some expectation somewhere that conditions will stay around for API compatibility sake.

Reply:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

"Once defined, the meaning of a Condition can not be changed arbitrarily - it becomes part of the API, and has the same backwards- and forwards-compatibility concerns of any other part of the API."

Reply:
Works for me.

Reply:
Done.